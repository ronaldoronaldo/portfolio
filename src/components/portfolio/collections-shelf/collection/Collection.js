import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  StyledCollection,
  Cover,
  CollectionNameBox,
  CollectionName, OpacityMask
} from './Collection.styles'

export class Collection extends Component {
  static propTypes = {
    collection: PropTypes.shape({
      name: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired
    }).isRequired,
  }

  static defaultProps = {
    collection: {
      name: '',
      image: ''
    }
  }

  handleCollectionClick = () => {
    console.log('collection clicked')
  }

  render() {
    const { collection, loading, ...rest } = this.props

    return (
      <StyledCollection {...rest}>
        <Cover
          onClick={() => this.handleCollectionClick()}
        >
          <OpacityMask/>
          <img src={'https://s3-sa-east-1.amazonaws.com/files.arvoredelivros.com.br/arvore-library-assets/arvore_colecoes_banner_' + collection.image + '.png'} alt={name} />
          <CollectionNameBox>
            <CollectionName>{collection.name}</CollectionName>
          </CollectionNameBox>
        </Cover>
      </StyledCollection>
    )
  }
}

export default Collection
