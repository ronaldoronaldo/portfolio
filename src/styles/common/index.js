import {
  WrapperStyle,
  HeaderStyle,
  BodyStyle,
  InputStyle,
  ColumnStyle,
  FinishButtons,
  IconButtonStyled,
  PagedListHeader,
  ContainerPlatform,
  LeftColumnStyle,
  RightColumnStyle,
  ResetPasswordButton,
  DisabledInputStyle,
  HeaderLeftWrapper,
  HeaderRightWrapper
} from './Common.styles'

export {
  WrapperStyle,
  HeaderStyle,
  BodyStyle,
  InputStyle,
  ColumnStyle,
  FinishButtons,
  IconButtonStyled,
  PagedListHeader,
  ContainerPlatform,
  LeftColumnStyle,
  RightColumnStyle,
  ResetPasswordButton,
  DisabledInputStyle,
  HeaderLeftWrapper,
  HeaderRightWrapper
}
