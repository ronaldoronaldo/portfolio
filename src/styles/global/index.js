import GlobalStyle from './Global.styles'

export { GlobalStyle }