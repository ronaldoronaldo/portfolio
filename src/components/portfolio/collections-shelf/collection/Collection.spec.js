import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import { Collection } from './Collection'
import { Cover } from './Collection.styles'
import { mount } from 'enzyme'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
const store = createStore(() => ({}))

describe('Collection', () => {
  let props

  beforeEach(() => {
    props = {
      collection: { name: '', image: '' },
      onClickCollection: jest.fn()
    }
  })

  it('should render correctly', () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <Collection {...props} />
        </MemoryRouter>
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it('calls click event', () => {
    const cover = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Collection {...props} />
        </MemoryRouter>
      </Provider>
    ).find(Cover)

    cover.simulate('click')
    expect(props.onClickCollection).toHaveBeenCalled()
  })
})
