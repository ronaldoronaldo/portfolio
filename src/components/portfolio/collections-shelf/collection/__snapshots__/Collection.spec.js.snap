// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection should render correctly 1`] = `
.c1 {
  position: relative;
  padding: 0 10px;
  width: 50%;
  max-width: 202px;
  height: auto;
  margin-bottom: 30px;
}

.c1 > a:first-child {
  border-radius: px;
  display: block;
}

.c0 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 72px;
  position: relative;
  top: -13px;
  background-color: hsl(220,69%,97%);
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
}

.c4 {
  font-size: 16px;
  color: hsl(220,2%,35%);
  text-align: center;
  max-width: 146px;
  font-weight: 700;
}

.c2 {
  width: 100%;
  height: auto;
  overflow: hidden;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  max-height: 270px;
}

.c2 > img {
  position: relative;
  left: -25%;
  width: 150%;
  vertical-align: top;
}

@media (min-width:768px) {
  .c1 {
    width: 202px;
    padding: 0px 10px 40px 10px;
    margin-bottom: 0;
    cursor: pointer;
  }
}

@media (max-width:768px) {
  .c1 {
    padding: 0px 8px 0 8px;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 18px;
    max-width: 160px;
  }
}

@media (min-width:1600px) {
  .c4 {
    font-size: 20px;
    max-width: 180px;
  }
}

<div
  className="c0 c1"
>
  <a
    className="c2"
    href="/biblioteca/colecao/:name"
    onClick={[Function]}
  >
    <img
      alt="nodejs"
      src="https://s3-sa-east-1.amazonaws.com/files.arvoredelivros.com.br/arvore-library-assets/arvore_colecoes_banner_.png"
    />
    <div
      className="c3"
    >
      <span
        className="c4"
      >
        
      </span>
    </div>
  </a>
</div>
`;
